### PR TITLE
Add fileuploader example to README

### PR DESCRIPTION
**Pull request by krisis**

### CODE REVIEW

File: examples/fileuploader.hs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env stack
```
Comment (harshavardhana) [inline examples/fileuploader.hs]: Add a minio copyright header.

File: README.md [hunk 2]
```diff
@@ -41,6 +41,74 @@ stack haddock
...
+
+  -- Parse command line argument, namely --filename.
+  filepath <- execParser cmdParser
+  let object = T.pack $ takeBaseName filepath
+
+  res <- runResourceT $ runMinio minioPlayCI $ do
+    -- Make a bucket; catch bucket already exists exception if thrown.
+    (MC.catch :: Minio () -> (MError -> Minio ()) -> Minio ())
+      (makeBucket bucket Nothing)
+      (\_ -> liftIO $ print "Bucket already exists, proceeding with upload file.")
```
Comment (krisis) [inline README.md]: The exception handler doesn't specifically catch `BucketExists` exception alone. The library needs to provide a mechanism to identify specific S3 errors. One approach is to provide combinators for all the common S3 [error codes](http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList).
E.g,
```haskell
-- rethrows exception if error code doesn't match "BucketAlreadyExists".
errBucketAlreadyExists :: MError -> Bool
```

File: examples/fileuploader.hs (hunk 1 above)
Comment (krisis) [inline examples/fileuploader.hs]: We need to add copyright header to all source code files. This will be addressed in a separate PR fixing #16

File: examples/fileuploader.hs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  -- Parse command line argument, namely --filename.
+  filepath <- execParser cmdParser
+  let object = pack $ takeBaseName filepath
+
+  res <- runResourceT $ runMinio minioPlayCI $ do
+    -- Make a bucket; catch bucket already exists exception if thrown.
+    (catch :: Minio () -> (MError -> Minio ()) -> Minio ())
+      (makeBucket bucket Nothing)
+      (\_ -> liftIO $ print "Bucket already exists, proceeding with upload file.")
+
```
Comment (donatello) [inline examples/fileuploader.hs]: After adding ScopedTypeVariables, we can simply do:

```haskell
    catch
      (makeBucket bucket Nothing)
      (\(_ :: MError) -> liftIO $
        putStrLn "Bucket already exists, proceeding with upload file.")
```

I think looks more readable.
